eventlog/common.go

```diff
@@ -0,0 +1,558 @@
+	0x00000009: "EV_CPU_MICROCODE",
+	0x0000000A: "EV_PLATFORM_CONFIG_FLAGS",
+	0x0000000B: "EV_TABLE_OF_DEVICES",
+	0x0000000C: "EV_COMPACT_HASH",
+	0x0000000D: "EV_IPL",
+	0x0000000E: "EV_IPL_PARTITION_DATA",
+	0x0000000F: "EV_NONHOST_CODE",
+	0x00000010: "EV_NONHOST_CONFIG",
+	0x00000011: "EV_NONHOST_INFO",
+	0x00000012: "EV_OMIT_BOOT_DEVICE_EVENTS",
+	0x80000000: "EV_EFI_EVENT_BASE",
+	0x80000001: "EV_EFI_VARIABLE_DRIVER_CONFIG",
+	0x80000002: "EV_EFI_VARIABLE_BOOT",
+	0x80000003: "EV_EFI_BOOT_SERVICES_APPLICATION",
+	0x80000004: "EV_EFI_BOOT_SERVICES_DRIVER",
+	0x80000005: "EV_EFI_RUNTIME_SERVICES_DRIVER",
+	0x80000006: "EV_EFI_GPT_EVENT",
+	0x80000007: "EV_EFI_ACTION",
+	0x80000008: "EV_EFI_PLATFORM_FIRMWARE_BLOB",
+	0x80000009: "EV_EFI_HANDOFF_TABLES",
+	0x8000000A: "EV_EFI_PLATFORM_FIRMWARE_BLOB2",
+	0x8000000B: "EV_EFI_HANDOFF_TABLES2",
+	0x8000000C: "EV_EFI_VARIABLE_BOOT2",
+	0x80000010: "EV_EFI_HCRTM_EVENT",
+	0x800000E0: "EV_EFI_VARIABLE_AUTHORITY",
+	0x800000E1: "EV_EFI_SPDM_FIRMWARE_BLOB",
+	0x800000E2: "EV_EFI_SPDM_FIRMWARE_CONFIG",
+	0x401:      "PCR_MAPPING",
+	0x402:      "HASH_START",
+	0x403:      "COMBINED_HASH",
+	0x404:      "MLE_HASH",
+	0x40a:      "BIOSAC_REG_DATA",
+	0x40b:      "CPU_SCRTM_STAT",
+	0x40c:      "LCP_CONTROL_HASH",
+	0x40d:      "ELEMENTS_HASH",
+	0x40e:      "STM_HASH",
+	0x40f:      "OSSINITDATA_CAP_HASH",
+	0x410:      "SINIT_PUBKEY_HASH",
+	0x411:      "LCP_HASH",
+	0x412:      "LCP_DETAILS_HASH",
+	0x413:      "LCP_AUTHORITIES_HASH",
+	0x414:      "NV_INFO_HASH",
+	0x416:      "EVTYPE_KM_HASH",
+	0x417:      "EVTYPE_BPM_HASH",
+	0x418:      "EVTYPE_KM_INFO_HASH",
+	0x419:      "EVTYPE_BPM_INFO_HASH",
+	0x41a:      "EVTYPE_BOOT_POL_HASH",
+	0x4ff:      "CAP_VALUE",
+}
+
+var evLogInfo *eventLogInfo
+
+// InitializeEventLogInfo - Function to create and intialize eventLogInfo variable with default values
+func initializeEventLogInfo() *eventLogInfo {
+	log.Trace("eventlog/common:initializeEventLogInfo() Entering")
+	defer log.Trace("eventlog/common:initializeEventLogInfo() Leaving")
+
+	eventLog := eventLogInfo{}
+	return &eventLog
+}
+
+// GetEventLogInfo - Getter function for eventLogInfo variable
+func getEventLogInfo() *eventLogInfo {
+	log.Trace("eventlog/common:getEventLogInfo() Entering")
+	defer log.Trace("eventlog/common:getEventLogInfo() Leaving")
+
+	if evLogInfo == nil {
+		evLogInfo = initializeEventLogInfo()
+	}
+	return evLogInfo
+}
+
+// ParseEventLog - Function to parse event log data from buffer
+func parseEventLog(buf *bytes.Buffer, size uint32) error {
+	log.Trace("eventlog/common:parseEventLog() Entering")
+	defer log.Trace("eventlog/common:parseEventLog() Leaving")
+
+	// Skip TCG_PCR_EVENT(Intel TXT spec. ver. 16.2) log data
+	tcgPcrEvent := tcgPcrEventV1{}
+	err := binary.Read(buf, binary.LittleEndian, &tcgPcrEvent.PcrIndex)
+	if err != nil {
+		return errors.Wrap(err, "eventlog/common:parseEventLog() There is an error reading binary data from buffer")
```

Comment:
Update the error messages so they are not all the same (this helps troubleshooting customer issues).  For example, "There was an error reading the PCR Index from the buffer" or "There was an error reading the Event Type from the buffer").